pysprint_car.py

```diff
@@ -1219,18 +1219,25 @@ def detect_crash(self, track: pysprint_tracks.Track):
                 return False
 
     def detect_spills(self, track: pysprint_tracks.Track):
-        if track.display_oil_spill and track.on_bridge_oil_spill==self.is_on_ramp_or_bridge():
-            if self.test_spill(pysprint_tracks.oil_spill_mask,track.oil_spill_position):
+
+        # oil spills
+        for spill in track.oil_spills:
+            #if track.display_oil_spill and track.on_bridge_oil_spill==self.is_on_ramp_or_bridge():
```

Comment:
@shazz: why did you comment `  if track.display_oil_spill and track.on_bridge_oil_spill==self.is_on_ramp_or_bridge():`?

It's essential to ensure obstacles are activated only if the car and obstacle are on the same level (both are on a bridge or both are below).

this does not apply to the tornado